openmdao/components/eq_constraints_comp.py

```diff
@@ -0,0 +1,230 @@
+        add_constraint : bool
+            Specifies whether to add an equality constraint.
+        **kwargs : dict
+            Additional arguments to be passed for the creation of the output variable.
+        """
+        super(EqualityConstraintsComp, self).__init__()
+        self._output_vars = {}
+        if name is not None:
+            self.add_eq_output(name, eq_units, lhs_name, rhs_name, rhs_val,
+                               use_mult, mult_name, mult_val, add_constraint, **kwargs)
+
+    def setup(self):
+        """
+        Define the independent variables, output variables, and partials.
+        """
+        for name, options in iteritems(self._output_vars):
+
+            for s in ('lhs', 'rhs', 'mult'):
+                if options['{0}_name'.format(s)] is None:
+                    options['{0}_name'.format(s)] = '{0}:{1}'.format(s, name)
+
+            val = options['kwargs'].get('val', np.ones(1))
+            if isinstance(val, Number):
+                n = 1
+            else:
+                n = len(val)
+            self._output_vars[name]['size'] = n
+
+            self.add_output(name, **options['kwargs'])
+
+            self.add_input(options['lhs_name'],
+                           val=np.ones(n),
+                           units=options['eq_units'])
+
+            self.add_input(options['rhs_name'],
+                           val=options['rhs_val'] * np.ones(n),
+                           units=options['eq_units'])
+
+            if options['use_mult']:
+                self.add_input(options['mult_name'],
+                               val=options['mult_val'] * np.ones(n),
+                               units=None)
+
+            self._scale_factor = np.ones(n)
+            self._dscale_drhs = np.ones(n)
+
+            ar = np.arange(n)
+            self.declare_partials(of=name, wrt=options['lhs_name'], rows=ar, cols=ar, val=1.0)
+            self.declare_partials(of=name, wrt=options['rhs_name'], rows=ar, cols=ar, val=1.0)
+
+            if options['use_mult']:
+                self.declare_partials(of=name, wrt=options['mult_name'], rows=ar, cols=ar, val=1.0)
+
+            if options['add_constraint']:
+                self.add_constraint(name, equals=0.)
+
+    def compute(self, inputs, outputs):
+        """
+        Calculate the output for each equality constraint.
+
+        Parameters
+        ----------
+        inputs : Vector
+            unscaled, dimensional input variables read via inputs[key]
+        outputs : Vector
+            unscaled, dimensional output variables read via outputs[key]
+        """
+        for name, options in iteritems(self._output_vars):
+            lhs = inputs[options['lhs_name']]
+            rhs = inputs[options['rhs_name']]
+
+            # Indices where the rhs is near zero or not near zero
+            idxs_nz = np.where(np.abs(rhs) < 2)[0]
+            idxs_nnz = np.where(np.abs(rhs) >= 2)[0]
+
+            # Compute scaling factors
```

Comment:
I don't really understand this scaling stuff, but it isn't what I would expect if I pulled an equality constraint comp off the shelf. 
1. It should at the very least be optional, if I don't want to use it. 
2. Don't understand why we are looking for points greater than 2, and less than 2, as our definition of near-zero. Might also need this to be user controlled value in options.

Reply:
Talked to Justin and understand this code now. We can leave it in as standard.

Reply:
left implementation same as BalanceComp, for future story...